Changelog for 10.23.0 (#16555)
MetaMask/metamask-extension: CHANGELOG.md

## [Unreleased]

## [10.23.0]
### Added
- Add Improved Token Allowance toggle on Experimental Settings ([#16291](https://github.com/MetaMask/metamask-extension/pull/16291))
- Add "What's New" Notification for Improved Token Allowance experience ([#16465](https://github.com/MetaMask/metamask-extension/pull/16465))
- Add Picker Network Component ([#16340](https://github.com/MetaMask/metamask-extension/pull/16340))
- Add Button component, unifying primary, secondary and link buttons([#16305](https://github.com/MetaMask/metamask-extension/pull/16305))
- Add Button Icon component ([#16277](https://github.com/MetaMask/metamask-extension/pull/16277))
- Swaps: enable Swaps functionality on Arbitrum and Optimism networks ([#16396](https://github.com/MetaMask/metamask-extension/pull/16396))

> - [FLASK] Catch and display errors in snaps insight ([#16416](https://github.com/MetaMask/metamask-extension/pull/16416))

Author: ty! done!

- [FLASK] Add snap cronjobs ([#16239](https://github.com/MetaMask/metamask-extension/pull/16239))

### Changed
- Replace every Address value by the Address component on SignTypedData v4 Signature screen ([#16018](https://github.com/MetaMask/metamask-extension/pull/16018))
- Update Address component on Transaction data screen by displaying Account name, Contact name, or Contract name when corresponds ([#15888](https://github.com/MetaMask/metamask-extension/pull/15888))
- Update Token Allowance screen for Improved Token Allowance experience ([#16157](https://github.com/MetaMask/metamask-extension/pull/16157))
- Bump `@metamask/providers` from `10.0.0` to `10.2.0` ([#16361](https://github.com/MetaMask/metamask-extension/pull/16361))

> - [FLASK] `snaps-monorepo@0.23.0` ([#16394](https://github.com/MetaMask/metamask-extension/pull/16394))

Reviewer: ```suggestion
- [FLASK] **BREAKING**: Snaps no longer automatically receive a `Buffer` polyfill ([#16394](https://github.com/MetaMask/metamask-extension/pull/16394))
  - To work around this you can either use typed arrays or include a polyfill yourself.
- [FLASK] **BREAKING**: Snap RPC methods now use `@metamask/key-tree@6.0.0` ([#16394](https://github.com/MetaMask/metamask-extension/pull/16394))
  - In the new version, all hexadecimal values are prefixed with `0x`
  - All fields containing the word `Buffer` has also been renamed to `Bytes`
  - Please update your snap to use the latest version 
  
```

Reviewer: Fine to keep as "changed" 😄

### Fixed
- Fix Settings Search pointing into the incorrect row for Token Detection entry ([#16407](https://github.com/MetaMask/metamask-extension/pull/16407))
- Fix Balance not updating when using a duplicated `chainId` network ([#14245](https://github.com/MetaMask/metamask-extension/pull/14245))

> - [FLASK] Fix an issue with updating snaps that have caveat permissions ([#16473](https://github.com/MetaMask/metamask-extension/pull/16473))

Reviewer: ```suggestion
- [FLASK] Fix an issue with updating snaps that require certain permissions ([#16473](https://github.com/MetaMask/metamask-extension/pull/16473))
```

> - [FLASK] Fix some issues with installing snaps that request eth_accounts ([#16365](https://github.com/MetaMask/metamask-extension/pull/16365))

Reviewer: ```suggestion
- [FLASK] Fix some issues with installing snaps that request `eth_accounts` ([#16365](https://github.com/MetaMask/metamask-extension/pull/16365))
```
